README.md

```diff
@@ -8,16 +8,207 @@ Local.
 
-## Key features
+For example:
+
+```
+// original document
+{
+  "id": ...,
+  "_rev": ...,
+  "prop1": "1",
+  "prop2": "2",
+}
+
+// data to be updated
+ds.updateOrCreate('User', {
+  prop1: 'updated1',
+}, function (err, res) {});
+
+// document after update
+{
+  "id": ...,
+  "_rev": ...,
+  "prop1": "updated1",
+}
+```
+
+*Please note how property `prop2` was completely dropped upon update.*
+
+Workaround: Do not pass partial values for the data object to be updated. If there are properties that are not being updated, please include the old value to keep data persistent.
+
+We have some discussion on update methods, the issue link can be found in [Feature Backlog](https://github.com/strongloop/loopback-connector-cloudant##feature-backlog) section
+
+### Careful Frequent Modification
+
+Cloudant is not designed to change same document frequently and multiple times. It stores status change by creating different documents and including the same unique id to tell that they are attached to the same item, not updating the same document.
+
+By modeling the data in separate documents that are only written once, we can ensure that documents never become conflicted, such as during concurrent access to the same document by separate processes.
+
+While it doesn't mean modification will cause trouble, by properly controlling the conflict, developer can still do a safe modify. For details, refer to [Conflict Control](https://github.com/strongloop/loopback-connector-cloudant#conflict-control)
+
+### Conflict Control
+
+TBD. 
+
+The basic idea is when modifying a document, user needs to control conflict by handling the revision of a document, currently the connector controls this process, after retriving the latest revision, connector uses it to update/delete doc, and returns 409 conflict error if doc changes during that time slot. In the middle, user could not interfere and provide their own conflict solution.
+
+This part will be done after we complete the `_rev` property refactor.
+
+## Model 
+
+### Map between model and document
+
+Unlike relational db or mongodb, Cloudant doesn't have a concept as 'table' or 'collection', data in a Cloudant database are all stored as documents.
+
+The connector uses a design document to represent a Loopback model, and common documents to represent model instances.
+
+The following is a comparison among different databases:
+
+|               | Model                        | Model Property                        | Model Instance         |
+|---------------|------------------------------|---------------------------------------|------------------------|
+| Relational DB | table                        | column in table                       | row in table           |
+| Mongodb       | collection                   | createIndex if property.index is true | document in collection |
+| Cloudant      | Design documents in database | NOT stored in document                     | common documents in database  |
```

Comment:
hmm, sorry I don't quite understand "If you know where model property is stored for cloudant", for example, where is it stored? could you give me a concrete example?